include/toolchain/xcc.h

```diff
@@ -36,4 +36,12 @@
 
 #endif /* __GCC_LINKER_CMD__ */
 
+#define __builtin_unreachable() __ASSERT(0, "Unreachable code")
+
+/* TODO: XCC doesn't define the below macros which are useful for checking
+ * overflows. This needs to be fixed.
+ */
+#define __builtin_add_overflow(a, b, output) 0
```

Comment:
The operation must be performed still, otherwise `output` won't be set and things won't work as expected:

```C
#define __builtin_add_overflow(a, b, output)   ({ *output = (a) + (b); 0; })
#define __builtin_mull_overflow(a, b, output)   ({ *output = (a) * (b); 0; })
```

Reply:
done.